### DISCUSSION TITLE
Getting the token from `~/.authinfo`

### DISCUSSION THREAD

**Discussion by magthe** [emacs-openai/openai#27] [General]
This what I've done to keep the OpenAI token in my `~/.authinfo.gpg` file.

This line goes into `~/.authinfo.gpg`:

```
machine api.openai.com login <org> password <pwd>
```

(My account is in multiple organisations, both my personal, my employer's and some client orgs too.)

The function extracting a token looks like this

```
(defun openai-key-auth-info ()
  (when-let ((auth-info (auth-source-search :host "api.openai.com" :login "<org>"))
	     (secret-fn (plist-get (car auth-info) :secret)))
    (funcall secret-fn)))
```

And finally

```
(setq openai-key #'openai-key-auth-info)
```

**Comment by jcs090218** [comment]
I'm sorry. So do you have an issue with the [auth-source](https://www.gnu.org/software/emacs/manual/html_mono/auth.html)? 🤔

**Comment by magthe** [comment]
I did, at first I didn't realise that `openai-key-auth-info` actually used `auth-source`, based on the documentation I tried with `https://api.openai.com/v1` but that didn't work so I read the code and realised I should only put in the hostname from the URL, not the whole URL. However, it only supports a single token per URL.

**Comment by jcs090218** [comment]
Got it! Sorry for the confusion! Is there anything you want us to improve?

**Comment by magthe** [comment]
No, I just wanted to share. I might try to add something to the README to make it a little more friendly to beginners like me.

**Comment by jcs090218** [comment]
Let me know if you need any help! :)
